### PR TITLE
Add support for Gyro tracking, button events, read calibration data from SPI flash

### PR DESCRIPTION
**Pull request by pbsds**
This PR add the following:

* Add ability to read from SPI flash
  * This is used to read the gyroscope and accelerometer calibration data, and the joycon color data
* Add a wrapper class `PythonicJoyCon` which provides a more pythonic interface
  * I would like to merge this into the main class
* Import all the classes to `__init__.py`, resulting in a more pythonic import convention
* Add support for listing multiple devices i `device.py`
  * meaning you could use more than a single pair of joycons
* Add `ButtonEventJoyCon` which tracks the change in button states and can provide events
* Add `GyroTrackingJoyCon` which tracks the current orientation of the joycon
* Fix some bugs
* Update the readme and docstrings
* Add some minor speedups

This change breaks the old interface to device.py, which is why i have bumped the version number

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -1,23 +1,32 @@
...
-    version='0.1.2',
+    version=version,
     description='Python driver for Nintendo Switch Joy-Con',
     long_description=readme,
     long_description_content_type='text/markdown',
-    author='tokoroten-lab, atsukoba',
-    author_email='tokoroten.lab@gmail.com, atsuya_kobayashi@yahoo.co.jp',
+    author='tokoroten-lab, atsukoba, pbsds',
+    author_email=', '.join([
+        'tokoroten.lab@gmail.com'
```
Comment (pbsds) [inline setup.py:22]:
```suggestion
        'tokoroten.lab@gmail.com',
```

**Comment by tokoroten-lab** [comment]
@pbsds Please check requested changes...

**Comment by h0m35t4r** [comment]
Great work on checking for vendor and product ID. May I add that my pair of Joy-Cons report the following:
**Vendor ID:** 1406
**Joy-Con Left product ID:** 8198
**Joy-Con Right Product ID:** 8199
They are the original grey pair that came with the console at launch.

File: pyjoycon/joycon.py [hunk 2]
```diff
@@ -60,27 +63,105 @@ def _open(self, vendor_id, product_id, serial):
...
+        return report[13] & 0x80, report[13:]  # (ack, data)
+
+    def _spi_flash_read(self, address, size) -> bytes:
+        assert size <= 0x1d
+        argument = address.to_bytes(4, "little") + size.to_bytes(1, "little")
+        ack, report = self._send_subcmd_get_response(b'\x10', argument)
+        if not ack:
+            raise IOError("After SPI read @ {address:#06x}: got NACK")
+
+        if report[:2] != b'\x90\x10':
```
[the remark this answers is not in the document]
Comment (pbsds) [reply pyjoycon/joycon.py:106]: Done :)

**Comment by pbsds** [comment]
Merging/rebasing/cherry-picking bca4f39983776610aeda3bd0980c3217cb78169b should not be neccesary, since it should get fixed when merged with `master`

I'm sorry for the delay. School caught up to me.

**Pull request merged**